x/wasm/types/authz.go

```diff
@@ -67,11 +78,20 @@ func (a StoreCodeAuthorization) ValidateBasic() error {
 	if len(a.Grants) == 0 {
 		return ErrEmpty.Wrap("grants")
 	}
-	for i, v := range a.Grants {
-		if err := v.ValidateBasic(); err != nil {
+	for i := 0; i < len(a.Grants); i++ {
+		if strings.EqualFold(string(a.Grants[i].CodeHash), codehashWildcard) && len(a.Grants) > 1 {
+			return sdkerrors.ErrInvalidRequest.Wrap("cannot have multiple grants when wildcard grant is one of them")
+		}
+		if err := a.Grants[i].ValidateBasic(); err != nil {
 			return errorsmod.Wrapf(err, "position %d", i)
 		}
+		for j := i + 1; j < len(a.Grants); j++ {
+			if strings.EqualFold(string(a.Grants[i].CodeHash), string(a.Grants[i].CodeHash)) {
```

Comment:
This is always true. One of the grants should be found by `j`. 
Alternatively, you can store the codehashes in a map and compare the len to be equal len(grants)